docs/rst/microros_documentation/benchmarking/benchmarking.rst

```diff
@@ -0,0 +1,169 @@
+
+The total memory (static plus stack plus dynamic) occupied is summarized in the plots below:
+
+.. figure:: /rst/figures/micro/benchmarking/pub_sub_total.png
+    :align: center
+
+From this data, its concluded that a publisher takes a total of ~ 550 B meanwhile a subscriber uses ~ 600 B. There is virtually no difference between these two entities, as the memory pools of micro-ROS RMW are shared among all the entities participating in a given application.
+
+To get a better understanding of the memory usage, the same is provided data but broken down into its the different memory types used:
+
+.. figure:: /rst/figures/micro/benchmarking/pub_sub_divided.png
+    :align: center
+
+This shows that both the static and the dynamic memories change with the entity number, while the stack usage stays constant.
+
+Service/Client applications
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The same approach is used to measure service and clients applications for a ``example_interfaces/srv/AddTwoInts`` service kind.
+Notice that this time the total memory is shown along its individual types:
+
+.. figure:: /rst/figures/micro/benchmarking/client_service.png
+    :align: center
+
+As concluded on the previous section, the memory used is almost identical for a ~ 500 B usage by both entity kinds. Note that it is also virtually identical to the memory used by a publisher or subscriber application.
+
+RMW History
+^^^^^^^^^^^
+
+As explained before, the topic memory comes from the RMW history, which is formed by static memory pools defined on compilation time. 
+For a varying ``RMW_UXRCE_MAX_HISTORY`` between 1 and 10:
+
+.. figure:: /rst/figures/micro/benchmarking/rmw_history.png
+    :align: center
+
+As expected, the static memory used by each history slot equals the ``MTU * RMW_UXRCE_STREAM_HISTORY`` formula, which for this scenario: ``512 * 4 = 2048 B``. For more details on the middleware memory usage, check the :ref:`memory management tutorial <micro_ros_middleware_memory>`.
+
+Throughput
+----------
+
+On this section data throughput will be measured for different transports and topic sizes. To perform this test, a simple best effort publisher micro-ROS application sends variable ``std_msgs/msg/String`` for 5 seconds.
+
+The transport are divided based on their ``framing`` configuration. More details can be found on the :ref:`Custom Transports tutorial<tutorials_micro_custom_transports>`.
+
+Stream-oriented transports
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The tested stream oriented transports and their configuration are:
+
+- USB-CDC: 115200 baudrate.
+- Serial UART: 115200 baudrate.
+- TCP (AWS Secure Sockets) based on `Wi-Fi-Pmod-Expansion-Board <https://www.renesas.com/eu/en/products/microcontrollers-microprocessors/ra-cortex-m-mcus/wi-fi-pmod-expansion-board-80211bgn-24g-wi-fi-pmod-expansion-board>`_.
+
+  - PMOD baudrate: 460800 bauds.
+
+.. figure:: /rst/figures/micro/benchmarking/stream_transports.png
```

Comment:
Just a comment here, what about maintaining the Y axis equally for all three graphs and setting the scale to log? So comparison will be easier. Same for the next graph. 

If is does not look good, ignore this comment.

Reply:
Updated here: https://github.com/eProsima/vulcanexus/pull/105/commits/c721a09b1f5450e9f298d8738b5cb92f11d945b2

Its improves the visual comparison.